packages/ra-core/src/inference/assertions.js

```diff
@@ -0,0 +1,31 @@
+import parseDate from 'date-fns/parse';
+
+export const isNumeric = value => !isNaN(parseFloat(value)) && isFinite(value);
+export const valuesAreNumeric = values => values.every(isNumeric);
+
+export const isInteger = value => Number.isInteger(value);
+export const valuesAreInteger = values => values.every(isInteger);
+
+export const isBoolean = value => typeof value === 'boolean';
```

Comment:
What abount nullable booleans? :p

Reply:
the inference currently falls back to string for all nullable values. That's a good idea of evolution for v2!